plan/logical_plan_builder.go

```diff
@@ -1768,6 +1770,13 @@ func (b *planBuilder) buildSemiJoin(outerPlan, innerPlan LogicalPlan, onConditio
 			joinPlan.JoinType = SemiJoin
 		}
 	}
+	if b.TableHints() != nil {
```

Comment:
Could we put the declaration of `outerAlias` and `innerAlias` here?

Reply:
Agree.